libgalois/src/MemoryPolicy.cpp

```diff
@@ -24,25 +25,31 @@ namespace {
 
 void
 LogIt(const std::string& str, katana::MemoryPolicy::MemInfo* mem_info) {
-  auto scope = katana::GetTracer().StartActiveSpan(str);
-  scope.span().Log(
-      "mem_stats", {
-                       {"rss_gb", katana::ToGB(mem_info->rss_bytes)},
-                       {"oom_score", mem_info->oom_score},
-                       {"used_ratio", mem_info->used_ratio},
-                       {"active_gb", katana::ToGB(mem_info->active)},
-                       {"standby_gb", katana::ToGB(mem_info->standby)},
-                   });
+  katana::GetTracer().GetActiveSpan().Log(
+      str, {
+               {"rss_gb", katana::ToGB(mem_info->rss_bytes)},
+               {"available_gb", katana::ToGB(mem_info->available_bytes)},
+               {"oom_score", mem_info->oom_score},
+               {"used_ratio", mem_info->used_ratio},
+               {"standby_gb", katana::ToGB(mem_info->standby)},
+           });
 }
 
 }  // namespace
 
 void
-katana::MemoryPolicy::UpdateMemInfo(
-    MemInfo* mem_info, count_t active, count_t standby) const {
-  mem_info->active = active;
+katana::MemoryPolicy::LogMemoryStats(
+    const std::string& message, count_t standby) {
+  MemInfo mem_info;
+  UpdateMemInfo(&mem_info, standby);
+  LogIt(message, &mem_info);
+}
+
+void
+katana::MemoryPolicy::UpdateMemInfo(MemInfo* mem_info, count_t standby) const {
   mem_info->standby = standby;
   mem_info->rss_bytes = katana::ProgressTracer::ParseProcSelfRssBytes();
+  mem_info->available_bytes = AvailableMemoryBytes();
   mem_info->used_ratio = (double)mem_info->rss_bytes / physical();
   mem_info->oom_score = OOMScore();
```

Comment:
TIL about OOMScore!

Reply:
It is strictly inferior to `/sys/fs/cgroup/memory/memory.pressure_level` which gives a direct interface to the kernel's judgment of a process' level of memory pressure, but that interface seems deprecated.